Sming/Arch/Esp8266/options.json

```diff
@@ -6,5 +6,24 @@
                 "filename": "$(FLASH_INIT_DATA_VCC)"
             }
         }
+    },
+    "legacy": {
```

Comment:
@mikee47 Why we need a new memory layout for ESP8266 given the fact that now the partition table can be moved to a different location? Why not keep the current layout and not introduce a legacy layout? 
- OTA will function out of the box for ESP8266 applications
- The firmware ROMs for big flashes can be as big as possible AND use the same entry point address